utility/serial.cpp

```diff
@@ -92,6 +92,53 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  if (_mode == SERIAL_MODE_HARDWARE || _mode == SERIAL_MODE_HARDWARE_AND_SOFTWARE) {
+    bool valid = true;
+    if (baud != 9600) {
+      _soft_serial->end();
+      _soft_serial->begin(9600); // shipping bluesmirf with default of 9600
```

Comment:
This isn't going to work to enter command mode if the bluetooth is already set to a baud rate other than 9600, correct? (another way of asking the question - the bluetooth module remembers baud rates between boots, correct?)

I think to make this work well, you're going to have to go through multiple baud rates checking for the acknowledgement that you successfully entered command mode (usually something like `OK`) before you assume that you can send an AT command successfully.